.github/workflows/reusable-e2e.yml

```diff
@@ -0,0 +1,328 @@
+on:
+  workflow_call:
+    inputs:
+      commit:
+        # commit to run tests against
+        required: true
+        type: string
+      repository:
+        # repository to run tests against (possibly a fork of amplify-ui)
+        required: true
+        type: string
```

Comment:
Here's where we get the `ref` to run e2e tests against. This follows the recommended security practice: [link](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-action-instead-of-an-inline-script-recommended).